lexer.py

```
@@ -0,0 +1,100 @@
+        elif callable(valid):
+            while valid(self.get()):
+                continue
+        self.backup()
+        return self.position != start
+        
+    def run_lexer( self ) -> None:
+        self.state = lex_start
+        while self.state is not None:
+            self.state = self.state(self)
+
+def lex_start( lex: Lexer ):
```

Ruff (0.11.9)

Missing return type annotation for public function `lex_start`

Add return type annotation: `None`

(ANN201)